http/src/request/audit_reason.rs

```diff
@@ -0,0 +1,118 @@
+use std::fmt::{Display, Formatter, Result as FmtResult};
+
+pub trait AuditLogReason: private::Sealed {
+    fn reason(self, reason: impl Into<String>) -> Result<Self, AuditLogReasonError>
+    where
+        Self: Sized;
+}
+
+mod private {
+    use crate::request::prelude::*;
+
+    /// Sealed stops crates other crates implementing the trait.
+    pub trait Sealed {}
+    impl<'a> Sealed for CreateInvite<'a> {}
+    impl<'a> Sealed for DeleteInvite<'a> {}
+    impl<'a> Sealed for DeleteMessage<'a> {}
+    impl<'a> Sealed for DeleteMessages<'a> {}
+    impl<'a> Sealed for UpdateChannel<'a> {}
+    impl<'a> Sealed for CreateWebhook<'a> {}
+    impl<'a> Sealed for DeleteWebhook<'a> {}
+    impl<'a> Sealed for UpdateWebhook<'a> {}
+    impl<'a> Sealed for CreatePin<'a> {}
+    impl<'a> Sealed for DeleteChannel<'a> {}
+    impl<'a> Sealed for DeleteChannelPermissionConfigured<'a> {}
+    impl<'a> Sealed for DeletePin<'a> {}
+    impl<'a> Sealed for UpdateChannelPermissionConfigured<'a> {}
+    impl<'a> Sealed for CreateBan<'a> {}
+    impl<'a> Sealed for DeleteBan<'a> {}
+    impl<'a> Sealed for CreateGuildChannel<'a> {}
+    impl<'a> Sealed for CreateGuildPrune<'a> {}
+    impl<'a> Sealed for CreateEmoji<'a> {}
+    impl<'a> Sealed for DeleteEmoji<'a> {}
+    impl<'a> Sealed for UpdateEmoji<'a> {}
+    impl<'a> Sealed for CreateGuildIntegration<'a> {}
+    impl<'a> Sealed for DeleteGuildIntegration<'a> {}
+    impl<'a> Sealed for UpdateGuildIntegration<'a> {}
+    impl<'a> Sealed for UpdateGuildMember<'a> {}
+    impl<'a> Sealed for AddRoleToMember<'a> {}
+    impl<'a> Sealed for RemoveMember<'a> {}
+    impl<'a> Sealed for RemoveRoleFromMember<'a> {}
+    impl<'a> Sealed for CreateRole<'a> {}
+    impl<'a> Sealed for DeleteRole<'a> {}
+    impl<'a> Sealed for UpdateRole<'a> {}
+    impl<'a> Sealed for UpdateGuild<'a> {}
+}
+
+impl AuditLogReasonError {
+    /// The maximum audit log reason length in codepoints.
```

Comment:
This uses different wording from the error enum, this should be the same for consistency